tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/AuthorityTests.cs

```diff
@@ -113,6 +115,61 @@ public async Task AuthorityValidationTestWithFalseValidateAuthorityAsync()
                 .ConfigureAwait(false);
         }
 
+        [TestMethod]
+        [DataRow(true)]
+        [DataRow(false)]
+        public async Task UncommonAuthorityValidationTestAsync(bool validateAuthority)
+        {
+            LabResponse labResponse = await LabUserHelper.GetDefaultUserAsync().ConfigureAwait(false);
+            LabUser user = labResponse.User;
+
+            IPublicClientApplication pca = PublicClientApplicationBuilder
+                .Create(labResponse.App.AppId)
+                .WithInstanceDiscoevryEndpoint(false)
+                .WithAuthority("https://bogus.microsoft.com/common", validateAuthority)
+                .WithTestLogging()
+                .Build();
+
+            Trace.WriteLine("Acquire a token using a not so common authority alias");
+
+            HttpRequestException exception = await AssertException.TaskThrowsAsync<HttpRequestException>(() =>
+                 pca.AcquireTokenByUsernamePassword(
+                    s_scopes,
+                     user.Upn,
+                     user.GetOrFetchPassword())
+                     .ExecuteAsync())
+                .ConfigureAwait(false);
+
+            Assert.AreEqual("No such host is known.", exception.Message);
+            Assert.AreEqual(11001,((SocketException)exception.InnerException).ErrorCode);
+        }
+
+        [TestMethod]
+        [DataRow(true)]
+        [DataRow(false)]
+        public async Task CommonAuthorityValidationTestAsync(bool validateAuthority)
```

Comment:
I don't see the point in making actual AAD tests here. This should be a unit test